sdk/nodejs/dagger/connect.ts

```diff
@@ -0,0 +1,147 @@
+// WIP(TomChv): This file shall be renamed to something else
+import { Client } from './client';
+import { execa, execaCommandSync } from 'execa';
+import path from 'path';
+import axios from 'axios';
+
+/**
+ * Cloak binary name
+ */
+const CLOAK_BINARY = "cloak";
+
+/**
+ * ConnectOpts defines option used to run cloak
+ * in dev mode.
+ * Options are based on `dagger cloak` CLI.
+ */
+export interface ConnectOpts {
+	LocalDirs?: Record<string, string>;
+	Port?: number;
+	Workdir?: string;
+	ConfigPath?: string;
+}
+
+/**
+ * connect runs cloak GraphQL server and initializes a
+ * Dagger client.
+ * This implementation is based on the existing Go SDK.
+ */
+export async function connect(config: ConnectOpts): Promise<Client> {
+	// exit with error if we are not using the non-Cloak dagger binary (< 0.3.0)
+	await verifyCloakBinary();
+
+	// Create config with default values that may be overridden
+	// by config if values are set.
+	const _config: Required<ConnectOpts> = {
+		Workdir: process.env['DAGGER_WORKDIR'] || process.cwd(),
+		ConfigPath: process.env['DAGGER_CONFIG'] || './dagger.json',
+		Port: 8080,
+		// Set LocalDirs to {} so it's not null
+		LocalDirs: {},
+		...config
+	};
+
+	const args = buildCLIArguments(_config);
+
+	// Start Cloak server.
+	//
+	const serverProcess = execa(CLOAK_BINARY, args, {
+		stdio: "inherit",
+		cwd: _config.Workdir
+	});
+
+	// Wait for Cloak server to be ready.
+	await waitCloakServer(_config.Port)
+
+	// TODO(TomChv): How transform this part here? This function currently
```

Comment:
@slumbering I need your extra attention on this part, I'm waiting for your opinion before implementing this part